### ISSUE TITLE
Cypress should not allow selecting a disabled option within a disabled fieldset

### ISSUE THREAD

**Issue by SSmale** [cypress-io/cypress#5951]
### Current behavior:

If you disable a form group it is still possible to change the value of a select input inside

### Desired behavior:

the select input is disabled so should error telling you it is unchangeable, similar to the not visible error

### Steps to reproduce: (app code and test code)

```js
<fieldset _ngcontent-c15="" class="nopadding" disabled="">
  <div _ngcontent-c15="" class="form-group">
      <label _ngcontent-c15="" for="selectBox">Test Select</label>
      <select _ngcontent-c15="" class="form-control ng-valid ng-untouched ng-pristine" formcontrolname="selectBox" id="selectBox" ng-reflect-name="selectBox">
          <option _ngcontent-c15="" disabled="true" value="0: undefined" hidden="">
              Select Version
          </option>
          <option _ngcontent-c15="" value="1" ng-reflect-value="1">1</option>
          <option _ngcontent-c15="" value="2" ng-reflect-value="2">2</option>
          <option _ngcontent-c15="" value="3" ng-reflect-value="3">3</option>
          <option _ngcontent-c15="" value="4" ng-reflect-value="4">4</option>
          <option _ngcontent-c15="" value="5" ng-reflect-value="5">5</option>
          <option _ngcontent-c15="" value="6" ng-reflect-value="6">6</option>
          <option _ngcontent-c15="" value="7" ng-reflect-value="7">7</option>
          <option _ngcontent-c15="" value="8" ng-reflect-value="8">8</option>
          <option _ngcontent-c15="" value="9" ng-reflect-value="9">9</option>
          <option _ngcontent-c15="" value="10" ng-reflect-value="10">10</option>
      </select>
  </div>
</fieldset>
```

```js
cy.get('select')
  .filter('#selectBox')
  .select(3)
  .find(':selected')
  .should('contain', 0)
```

### Versions

cy: 3.5
os: windows 10
browser: chrome 78

**Comment by jennifer-shehane** [comment]
I see that you are using an older version of Cypress. Before I dive in to completely recreate the issue, could you update to the [current version of Cypress](https://on.cypress.io/changelog) and let me know if this is still happening for you? Your issue may have already been fixed. Thanks!

**Comment by SSmale** [comment]
Hi @jennifer-shehane,

I have tested this with version 3.8.0 downloaded this morning and the issue is still there.

**Comment by jennifer-shehane** [comment]
@SSmale Please provide the pure html + css + js to reproduce this. The below code does not produce the behavior you are saying. We'll have to close the issue without a way to run this. 

`index.html`

```
<!DOCTYPE html>
<html>
<body>
<fieldset _ngcontent-c15="" class="nopadding" disabled="">
  <div _ngcontent-c15="" class="form-group">
    <label _ngcontent-c15="" for="selectBox">Test Select</label>
    <select _ngcontent-c15="" class="form-control ng-valid ng-untouched ng-pristine" formcontrolname="selectBox"
      id="selectBox" ng-reflect-name="selectBox">
      <option _ngcontent-c15="" disabled="true" value="0: undefined" hidden="">
        Select Version
      </option>
      <option _ngcontent-c15="" value="1" ng-reflect-value="1">1</option>
      <option _ngcontent-c15="" value="2" ng-reflect-value="2">2</option>
      <option _ngcontent-c15="" value="3" ng-reflect-value="3">3</option>
      <option _ngcontent-c15="" value="4" ng-reflect-value="4">4</option>
      <option _ngcontent-c15="" value="5" ng-reflect-value="5">5</option>
      <option _ngcontent-c15="" value="6" ng-reflect-value="6">6</option>
      <option _ngcontent-c15="" value="7" ng-reflect-value="7">7</option>
      <option _ngcontent-c15="" value="8" ng-reflect-value="8">8</option>
      <option _ngcontent-c15="" value="9" ng-reflect-value="9">9</option>
      <option _ngcontent-c15="" value="10" ng-reflect-value="10">10</option>
    </select>
  </div>
</fieldset>
</body>
</html>
```

```js
it('selects', function () {
  cy.visit('index.html')
  cy.get('select')
    .filter('#selectBox')
    .select(3)
    .find(':selected')
    .should('contain', 0)
})
```

<img width="973" alt="Screen Shot 2019-12-17 at 4 26 22 PM" src="https://user-images.githubusercontent.com/1271364/70985114-01a13680-20ea-11ea-961a-4855793cc002.png">

**Comment by SSmale** [comment]
`it('selects', function () {
  cy.visit('index.html')
  cy.get('select')
    .filter('#selectBox')
    .select('3')
    .find(':selected')
    .should('contain', 0)
})`

the select should be a string not a number

**Comment by jennifer-shehane** [comment]
I can recreate this. This likely has to do with us not considering a fieldset that is disabled - that it will disable all selects within it.

```js
it('selects', () => { 
  cy.visit('index.html') 
  cy.get('select').find(':selected').should('contain', 1) 
  cy.get('select').select('3').find(':selected').should('contain', 1) 
})
```

```html
<html>
<body>
  <fieldset disabled="">
    <select>
      <option value="1">1</option>
      <option value="2">2</option>
      <option value="3">3</option>
    </select>
  </fieldset>
</body>
</html>
```

<img width="669" alt="Screen Shot 2020-10-20 at 4 12 33 PM" src="https://user-images.githubusercontent.com/1271364/96569406-185a8000-12ef-11eb-90a3-a384046df38a.png">

**Comment by jennifer-shehane** [comment]
Released in [`7.2.0`](https://on.cypress.io/changelog#7-2-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v7.2.0, please [open a new issue](https://github.com/cypress-io/cypress/issues/new/choose).
